django_graph_api/tests/graphql/test_types.py

```diff
@@ -0,0 +1,88 @@
+from django_graph_api.graphql.schema import Schema
+from django_graph_api.graphql.types import IntegerField, Object
+
+schema = Schema()
+
+
+# http://facebook.github.io/graphql/#sec-Scalars
+
+def test_coerce_integer_result_from_string():
```

Comment:
@melinath Was this the type of test you were thinking?

If you have any suggestions for simplifying these tests, please let me know. I was just focusing on getting them working and passing.

Reply:
I was thinking a little more low-level, i.e. something like:

```
def test_boolean_coerce():
    result = Boolean.coerce_result(1)
    assert result is True
```

And then just have tests running down all the possible combinations of valid/invalid values laid out in the spec for scalar values.

It would then also be good to have a test for Field.get_value() to ensure that it's passing the raw value to the type's coercion method. I think that would look something like:

```
def test_field_get_value_calls_coerce():
    field = Field()
    field.type_ = mock.Mock()
    field.obj = {'hi': 'lo'}
    field.name = 'hi'
    field.get_value()
    field.type_.coerce_result.assert_called_once_with('lo')
```

I'm pretty sure that's not quite right in several ways but I hope it helps!

Reply:
Having a more complete query test like test_hero_name_and_episodes is also 💯